File: app/src/main/java/io/treehouses/remote/bases/BaseBottomSheetDialog.java
package io.treehouses.remote.bases;

import android.content.Context;
import android.view.WindowManager;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;

import io.treehouses.remote.callback.HomeInteractListener;

public class BaseBottomSheetDialog extends BottomSheetDialogFragment {
    protected Context context;
    protected HomeInteractListener listener;

    @Override
    public void onAttach(Context c) {
        super.onAttach(c);
        try {
            listener = (HomeInteractListener) c;
            context = c;
        } catch (ClassCastException e) {
            throw new ClassCastException(c.toString() + " must implement HomeInteractListener");
        }
    }
}

File: app/src/main/java/io/treehouses/remote/bases/BaseBottomSheetDialog.kt
package io.treehouses.remote.bases

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import io.treehouses.remote.R
import io.treehouses.remote.callback.HomeInteractListener

open class BaseBottomSheetDialog : BottomSheetDialogFragment() {
    protected lateinit var listener: HomeInteractListener

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        setStyle(DialogFragment.STYLE_NORMAL, R.style.DialogStyle)
        return super.onCreateDialog(savedInstanceState)
    }

    override fun onAttach(c: Context) {
        super.onAttach(c)
        try {
            listener = c as HomeInteractListener
        } catch (e: ClassCastException) {
            throw ClassCastException("$c must implement HomeInteractListener")
        }
    }
}